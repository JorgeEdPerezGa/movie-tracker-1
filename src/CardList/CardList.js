import React, { Component } from 'react';
import helper from '../API/helper.js';
import { connect } from 'react-redux';
import { addMovies } from './CardListActions.js';
const { fetchMovies } = helper;

class CardList extends Component {
  

  async componentDidMount(){
    const getMovies = await fetchMovies();
  }
    addMovies(getMovies)

  render(){
    return(
      <div>hello</div>
    )
  }
}

const mapStateToProps = (store) => {
  return {
    movies: store.movies
  }
}


const mapDispatchToProps = (dispatch) => {
 return {
  addMovie: (getMovies) => {
    dispatch(addMovies(getMovies));
  }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CardList);
