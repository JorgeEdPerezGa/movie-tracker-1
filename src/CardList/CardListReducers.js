export const addMovies = (store = [], action) => {
  switch(action.type){
    case 'ADD_MOVIES':
    return [...action.addMovies]
    default:
    return store;
  }
}
