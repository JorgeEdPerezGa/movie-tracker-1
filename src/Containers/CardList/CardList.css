.card-container {
  margin-top: -100px;
  background: #1f1f1f;
  justify-content: center;
  display: flex;
  flex-wrap: wrap; }
