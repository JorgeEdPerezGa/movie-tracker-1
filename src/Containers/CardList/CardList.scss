.card-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-top: 90px;
  padding-bottom: 50px;
  margin-left: auto;
  margin-right: auto;
}

.card-container {
   justify-content: center;
    display: -ms-flexbox;
    display: flex;
    flex-wrap: wrap;
    margin-top: -140px;
    position: relative;
}

button{
    &:hover {
    cursor: pointer;
  } &:focus{
    outline: 0 !important;
  }
}
