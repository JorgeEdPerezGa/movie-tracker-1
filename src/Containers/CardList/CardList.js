import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../Components/Card/Card';
import './CardList.css'
import { addFavorite } from '../../Actions/Actions';
import { fetchFavorites } from '../../API/helper.js';


class CardList extends Component {
  constructor() {
    super();
    this.state = {
      favorites: []
    }
    this.addFavorite = this.addFavorite.bind(this);
  }

 addFavorite(card)  {
    this.setState({favorites: [...this.state.favorites, card]})
  }


  render() {
    const movieCards = this.props.movies.map((movie) => {
     return <Card
       key = {movie.title}
       title = {movie.title}
       overview = {movie.overview}
       poster = {movie.poster}
       vote = {movie.vote}
       backdrop = {movie.backdrop}
       favorite={movie.favorite}
       favorites={this.addFavorite}
     />
   })

    return(
      <div className = 'card-list'>
       {movieCards}
      </div>
    )
 }
}

const mapStateToProps = (store) => {
  return {
    movies: store.movies,
    favorites: store.favorite
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addFavorite: (favorite) => {
      dispatch(addFavorite(favorite));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CardList);
