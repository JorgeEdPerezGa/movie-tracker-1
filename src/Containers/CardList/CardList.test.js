import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

describe('CardList Tests', () => {

  it.skip('should be defined', () => {
    const renderedCardList = shallow(<CardList />);

    expect(renderedCardList).toBeDefined();
  })

})
