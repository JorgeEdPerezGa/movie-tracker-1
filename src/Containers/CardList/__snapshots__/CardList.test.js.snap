// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    favorites={
      Array [
        "array",
      ]
    }
    history={
      Object {
        "hola": "blah",
      }
    }
    location={
      Object {
        "pathname": "current-path",
      }
    }
    match={
      Object {
        "hola": "blah",
      }
    }
    movies={
      Array [
        "coco",
      ]
    }
    setFavorites={[Function]}
    user={
      Array [
        "blah",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Backdrop
          backdrop={undefined}
        />,
        <div
          className="card-container"
        >
          <Card
            addToFavorites={[Function]}
            backdrop={undefined}
            isFavorite={undefined}
            movie="coco"
            overview={undefined}
            poster={undefined}
            title={undefined}
            user={
              Array [
                "blah",
              ]
            }
            vote={undefined}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backdrop": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              addToFavorites={[Function]}
              backdrop={undefined}
              isFavorite={undefined}
              movie="coco"
              overview={undefined}
              poster={undefined}
              title={undefined}
              user={
                Array [
                  "blah",
                ]
              }
              vote={undefined}
            />,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "addToFavorites": [Function],
              "backdrop": undefined,
              "isFavorite": undefined,
              "movie": "coco",
              "overview": undefined,
              "poster": undefined,
              "title": undefined,
              "user": Array [
                "blah",
              ],
              "vote": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Backdrop
            backdrop={undefined}
          />,
          <div
            className="card-container"
          >
            <Card
              addToFavorites={[Function]}
              backdrop={undefined}
              isFavorite={undefined}
              movie="coco"
              overview={undefined}
              poster={undefined}
              title={undefined}
              user={
                Array [
                  "blah",
                ]
              }
              vote={undefined}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backdrop": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                addToFavorites={[Function]}
                backdrop={undefined}
                isFavorite={undefined}
                movie="coco"
                overview={undefined}
                poster={undefined}
                title={undefined}
                user={
                  Array [
                    "blah",
                  ]
                }
                vote={undefined}
              />,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "addToFavorites": [Function],
                "backdrop": undefined,
                "isFavorite": undefined,
                "movie": "coco",
                "overview": undefined,
                "poster": undefined,
                "title": undefined,
                "user": Array [
                  "blah",
                ],
                "vote": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
